rcl/include/rcl/time.h

```diff
@@ -45,340 +46,218 @@ typedef rcutils_time_point_value_t rcl_time_point_value_t;
 typedef rcutils_duration_value_t rcl_duration_value_t;
 
 /// Time source type, used to indicate the source of a time measurement.
-enum rcl_time_source_type_t
+typedef enum rcl_clock_type_t
 {
-  RCL_TIME_SOURCE_UNINITIALIZED = 0,
+  RCL_CLOCK_UNINITIALIZED = 0,
   RCL_ROS_TIME,
   RCL_SYSTEM_TIME,
   RCL_STEADY_TIME
-};
+} rcl_clock_type_t;
 
 /// Encapsulation of a time source.
-typedef struct rcl_time_source_t
+typedef struct rcl_clock_t
 {
-  enum rcl_time_source_type_t type;
+  enum rcl_clock_type_t type;
   void (* pre_update)(void);
   void (* post_update)(void);
   rcl_ret_t (* get_now)(void * data, rcl_time_point_value_t * now);
   // void (*set_now) (rcl_time_point_value_t);
   void * data;
-} rcl_time_source_t;
+} rcl_clock_t;
 
-struct rcl_ros_time_source_storage_t;
+struct rcl_ros_clock_storage_t;
```

Comment:
Does this need to be forward declared? I only see it being used in the `time.c` file.